dxr/lines.py

```diff
@@ -464,15 +464,11 @@ def finished_tags(lines, refs, regions):
     """
     # Plugins return unicode offsets, not byte ones.
 
-    # Get start and endpoints of intervals:
-    tags = list(tag_boundaries(chain(refs, regions)))
-
-    tags.extend(line_boundaries(lines))
-
-    # Sorting is actually not a significant use of time in an actual indexing
-    # run.
-    tags.sort(key=nesting_order)  # balanced_tags undoes this, but we tolerate
-                                  # that in html_lines().
+    # balanced_tags undoes the sorting, but we tolerate that in html_lines().
+    # Remark: this sort is the memory peak.
+    tags = sorted(chain(tag_boundaries(chain(refs, regions)),
```

Comment:
The reason I used sort() rather than sorted() was that I believe sorted() makes a copy of the collection being sorted, doubling RAM use. (This would have to be true, since sorted() works on any (possibly lazy) iterable, not just lists.)